{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_10x_mtx(\n",
    "    \"C:/Users/Vaishnavi Gaad/Desktop/MS/Sem 2/Capstone Project/Testing/rawData_human/rawData_human/countTable_human\",  # the directory with the `.mtx` file\n",
    "    var_names=\"gene_symbols\",  # use gene symbols for the variable names (variables-axis index)\n",
    "    cache=True,  # write a cache file for faster subsequent reading\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 272867 × 32738\n",
       "    var: 'gene_ids', 'feature_types'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGAGAGCTC-1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGCCGGTAA-1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGCGATTCT-1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGCTAACTC-1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGCTTTGGT-1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGAGTGCAGGT-41</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGGTGTGCCTG-41</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGTCCCATGGG-41</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGTCCGATGTA-41</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGTCGTAACAC-41</th>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>272867 rows × 0 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: [AAACCTGAGAGAGCTC-1, AAACCTGAGCCGGTAA-1, AAACCTGAGCGATTCT-1, AAACCTGAGCTAACTC-1, AAACCTGAGCTTTGGT-1, AAACCTGAGGATGGAA-1, AAACCTGAGTGTACCT-1, AAACCTGAGTTAACGA-1, AAACCTGCAAGTTCTG-1, AAACCTGCACGGTTTA-1, AAACCTGCAGGACGTA-1, AAACCTGCATAGAAAC-1, AAACCTGGTAAGCACG-1, AAACCTGGTATTAGCC-1, AAACCTGGTCGGCATC-1, AAACCTGGTCTCGTTC-1, AAACCTGGTTTGTTGG-1, AAACCTGTCAGGATCT-1, AAACCTGTCCAGGGCT-1, AAACCTGTCTCCAGGG-1, AAACCTGTCTCGCATC-1, AAACGGGAGACCTAGG-1, AAACGGGAGACTAGGC-1, AAACGGGAGCACAGGT-1, AAACGGGAGTCCCACG-1, AAACGGGAGTGTGGCA-1, AAACGGGAGTTAGGTA-1, AAACGGGCAATCTACG-1, AAACGGGCACCAGATT-1, AAACGGGCACGACGAA-1, AAACGGGCAGCCTATA-1, AAACGGGCAGCGAACA-1, AAACGGGCATTAACCG-1, AAACGGGGTAAACGCG-1, AAACGGGGTAGCGTGA-1, AAACGGGGTAGGGACT-1, AAACGGGGTGACTACT-1, AAACGGGGTGCACTTA-1, AAACGGGGTGCGGTAA-1, AAACGGGTCAGAGACG-1, AAACGGGTCAGTACGT-1, AAACGGGTCATTGCCC-1, AAACGGGTCCGCAAGC-1, AAACGGGTCCTCGCAT-1, AAACGGGTCGCCTGAG-1, AAAGATGAGAGCTGGT-1, AAAGATGAGGAATCGC-1, AAAGATGAGGACCACA-1, AAAGATGAGGACTGGT-1, AAAGATGAGGATGGTC-1, AAAGATGAGGGCTTCC-1, AAAGATGAGGGTCGAT-1, AAAGATGAGTCATGCT-1, AAAGATGAGTTAGGTA-1, AAAGATGCAAAGGAAG-1, AAAGATGCAAGACGTG-1, AAAGATGCAATAGAGT-1, AAAGATGCACTACAGT-1, AAAGATGCAGCCTGTG-1, AAAGATGCATTGGCGC-1, AAAGATGGTATCACCA-1, AAAGATGGTCCGTTAA-1, AAAGATGGTGTGAAAT-1, AAAGATGTCACATGCA-1, AAAGATGTCCCAACGG-1, AAAGATGTCGGATGTT-1, AAAGATGTCTCCTATA-1, AAAGCAAAGCACACAG-1, AAAGCAAAGGAACTGC-1, AAAGCAAAGGCAAAGA-1, AAAGCAAAGTGGAGTC-1, AAAGCAACACACGCTG-1, AAAGCAACACGGCCAT-1, AAAGCAACAGAGTGTG-1, AAAGCAACATCCTAGA-1, AAAGCAACATGCTGGC-1, AAAGCAACATTGGTAC-1, AAAGCAACATTTGCCC-1, AAAGCAAGTACCGAGA-1, AAAGCAAGTCTCATCC-1, AAAGCAAGTTACTGAC-1, AAAGCAAGTTCAGTAC-1, AAAGCAAGTTCATGGT-1, AAAGCAATCACTCCTG-1, AAAGCAATCCAACCAA-1, AAAGCAATCCTCAACC-1, AAAGCAATCCTCCTAG-1, AAAGCAATCGAATGGG-1, AAAGCAATCGGAAACG-1, AAAGCAATCTAACTCT-1, AAAGCAATCTAGCACA-1, AAAGCAATCTCTAGGA-1, AAAGCAATCTTTACAC-1, AAAGTAGAGCACCGCT-1, AAAGTAGAGCCGGTAA-1, AAAGTAGAGGTTACCT-1, AAAGTAGAGTACGCGA-1, AAAGTAGAGTAGCCGA-1, AAAGTAGCAACACGCC-1, AAAGTAGCAAGACGTG-1, ...]\n",
       "\n",
       "[272867 rows x 0 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "272867"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(adata.obs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "      <th>feature_types</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MIR1302-10</th>\n",
       "      <td>MIR1302-10</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAM138A</th>\n",
       "      <td>FAM138A</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OR4F5</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RP11-34P13.7</th>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RP11-34P13.8</th>\n",
       "      <td>RP11-34P13.8</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC145205.1</th>\n",
       "      <td>AC145205.1</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BAGE5</th>\n",
       "      <td>BAGE5</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CU459201.1</th>\n",
       "      <td>CU459201.1</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC002321.2</th>\n",
       "      <td>AC002321.2</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC002321.1</th>\n",
       "      <td>AC002321.1</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32738 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  gene_ids    feature_types\n",
       "MIR1302-10      MIR1302-10  Gene Expression\n",
       "FAM138A            FAM138A  Gene Expression\n",
       "OR4F5                OR4F5  Gene Expression\n",
       "RP11-34P13.7  RP11-34P13.7  Gene Expression\n",
       "RP11-34P13.8  RP11-34P13.8  Gene Expression\n",
       "...                    ...              ...\n",
       "AC145205.1      AC145205.1  Gene Expression\n",
       "BAGE5                BAGE5  Gene Expression\n",
       "CU459201.1      CU459201.1  Gene Expression\n",
       "AC002321.2      AC002321.2  Gene Expression\n",
       "AC002321.1      AC002321.1  Gene Expression\n",
       "\n",
       "[32738 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.var_names_make_unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<272867x32738 sparse matrix of type '<class 'numpy.float32'>'\n",
       "\twith 351710660 stored elements in Compressed Sparse Row format>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  (0, 67)\t1.0\n",
      "  (0, 89)\t1.0\n",
      "  (0, 121)\t1.0\n",
      "  (0, 166)\t4.0\n",
      "  (0, 189)\t1.0\n",
      "  (0, 229)\t1.0\n",
      "  (0, 231)\t2.0\n",
      "  (0, 248)\t1.0\n",
      "  (0, 249)\t1.0\n",
      "  (0, 326)\t1.0\n",
      "  (0, 412)\t1.0\n",
      "  (0, 464)\t1.0\n",
      "  (0, 492)\t6.0\n",
      "  (0, 549)\t1.0\n",
      "  (0, 556)\t4.0\n",
      "  (0, 592)\t1.0\n",
      "  (0, 610)\t2.0\n",
      "  (0, 617)\t1.0\n",
      "  (0, 688)\t1.0\n",
      "  (0, 742)\t1.0\n",
      "  (0, 761)\t1.0\n",
      "  (0, 764)\t1.0\n",
      "  (0, 799)\t1.0\n",
      "  (0, 819)\t1.0\n",
      "  (0, 859)\t1.0\n",
      "  :\t:\n",
      "  (272866, 32144)\t1.0\n",
      "  (272866, 32160)\t1.0\n",
      "  (272866, 32165)\t3.0\n",
      "  (272866, 32203)\t1.0\n",
      "  (272866, 32207)\t1.0\n",
      "  (272866, 32322)\t1.0\n",
      "  (272866, 32324)\t2.0\n",
      "  (272866, 32402)\t1.0\n",
      "  (272866, 32404)\t2.0\n",
      "  (272866, 32412)\t5.0\n",
      "  (272866, 32417)\t2.0\n",
      "  (272866, 32419)\t1.0\n",
      "  (272866, 32482)\t1.0\n",
      "  (272866, 32507)\t4.0\n",
      "  (272866, 32523)\t2.0\n",
      "  (272866, 32679)\t3.0\n",
      "  (272866, 32693)\t1.0\n",
      "  (272866, 32697)\t3.0\n",
      "  (272866, 32698)\t4.0\n",
      "  (272866, 32699)\t1.0\n",
      "  (272866, 32702)\t3.0\n",
      "  (272866, 32705)\t1.0\n",
      "  (272866, 32708)\t1.0\n",
      "  (272866, 32717)\t1.0\n",
      "  (272866, 32726)\t1.0\n"
     ]
    }
   ],
   "source": [
    "print(adata.X)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "sc.pp.normalize_total(adata, target_sum=1e4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  (0, 67)\t5.6980057\n",
      "  (0, 89)\t5.6980057\n",
      "  (0, 121)\t5.6980057\n",
      "  (0, 166)\t22.792023\n",
      "  (0, 189)\t5.6980057\n",
      "  (0, 229)\t5.6980057\n",
      "  (0, 231)\t11.396011\n",
      "  (0, 248)\t5.6980057\n",
      "  (0, 249)\t5.6980057\n",
      "  (0, 326)\t5.6980057\n",
      "  (0, 412)\t5.6980057\n",
      "  (0, 464)\t5.6980057\n",
      "  (0, 492)\t34.188034\n",
      "  (0, 549)\t5.6980057\n",
      "  (0, 556)\t22.792023\n",
      "  (0, 592)\t5.6980057\n",
      "  (0, 610)\t11.396011\n",
      "  (0, 617)\t5.6980057\n",
      "  (0, 688)\t5.6980057\n",
      "  (0, 742)\t5.6980057\n",
      "  (0, 761)\t5.6980057\n",
      "  (0, 764)\t5.6980057\n",
      "  (0, 799)\t5.6980057\n",
      "  (0, 819)\t5.6980057\n",
      "  (0, 859)\t5.6980057\n",
      "  :\t:\n",
      "  (272866, 32144)\t4.96771\n",
      "  (272866, 32160)\t4.96771\n",
      "  (272866, 32165)\t14.90313\n",
      "  (272866, 32203)\t4.96771\n",
      "  (272866, 32207)\t4.96771\n",
      "  (272866, 32322)\t4.96771\n",
      "  (272866, 32324)\t9.93542\n",
      "  (272866, 32402)\t4.96771\n",
      "  (272866, 32404)\t9.93542\n",
      "  (272866, 32412)\t24.83855\n",
      "  (272866, 32417)\t9.93542\n",
      "  (272866, 32419)\t4.96771\n",
      "  (272866, 32482)\t4.96771\n",
      "  (272866, 32507)\t19.87084\n",
      "  (272866, 32523)\t9.93542\n",
      "  (272866, 32679)\t14.90313\n",
      "  (272866, 32693)\t4.96771\n",
      "  (272866, 32697)\t14.90313\n",
      "  (272866, 32698)\t19.87084\n",
      "  (272866, 32699)\t4.96771\n",
      "  (272866, 32702)\t14.90313\n",
      "  (272866, 32705)\t4.96771\n",
      "  (272866, 32708)\t4.96771\n",
      "  (272866, 32717)\t4.96771\n",
      "  (272866, 32726)\t4.96771\n"
     ]
    }
   ],
   "source": [
    "print(adata.X)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.X.min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = pd.read_csv(\"C:/Users/Vaishnavi Gaad/Desktop/MS/Sem 2/Capstone Project/Testing/annot_humanAll.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "barcodes = metadata['cell']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 167598 × 32738\n",
       "    var: 'gene_ids', 'feature_types'"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abc=adata[barcodes].copy()\n",
    "abc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UMAP_1</th>\n",
       "      <th>UMAP_2</th>\n",
       "      <th>cluster</th>\n",
       "      <th>annot</th>\n",
       "      <th>sample</th>\n",
       "      <th>patient</th>\n",
       "      <th>digest</th>\n",
       "      <th>typeSample</th>\n",
       "      <th>diet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAACCTGTCAGGATCT-1</th>\n",
       "      <td>0.738062</td>\n",
       "      <td>-3.097089</td>\n",
       "      <td>34</td>\n",
       "      <td>Mono+mono derived cells</td>\n",
       "      <td>CISE06</td>\n",
       "      <td>H02</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>citeSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACGGGAGCACAGGT-1</th>\n",
       "      <td>0.367818</td>\n",
       "      <td>-3.251462</td>\n",
       "      <td>34</td>\n",
       "      <td>Mono+mono derived cells</td>\n",
       "      <td>CISE06</td>\n",
       "      <td>H02</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>citeSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACGGGAGTGTGGCA-1</th>\n",
       "      <td>-0.728939</td>\n",
       "      <td>-5.994781</td>\n",
       "      <td>14</td>\n",
       "      <td>Mono+mono derived cells</td>\n",
       "      <td>CISE06</td>\n",
       "      <td>H02</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>citeSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACGGGAGTTAGGTA-1</th>\n",
       "      <td>2.814624</td>\n",
       "      <td>-7.131484</td>\n",
       "      <td>7</td>\n",
       "      <td>Mono+mono derived cells</td>\n",
       "      <td>CISE06</td>\n",
       "      <td>H02</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>citeSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACGGGCACCAGATT-1</th>\n",
       "      <td>3.420414</td>\n",
       "      <td>-6.966918</td>\n",
       "      <td>7</td>\n",
       "      <td>Mono+mono derived cells</td>\n",
       "      <td>CISE06</td>\n",
       "      <td>H02</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>citeSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTACCAAGAGAGGTA-41</th>\n",
       "      <td>-3.674058</td>\n",
       "      <td>7.171934</td>\n",
       "      <td>41</td>\n",
       "      <td>Fibroblasts</td>\n",
       "      <td>CS171</td>\n",
       "      <td>H38</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>scRnaSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTACCAGTATCGAAA-41</th>\n",
       "      <td>-4.356690</td>\n",
       "      <td>7.339135</td>\n",
       "      <td>41</td>\n",
       "      <td>Fibroblasts</td>\n",
       "      <td>CS171</td>\n",
       "      <td>H38</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>scRnaSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTCCTCGTGAGGCAT-41</th>\n",
       "      <td>-3.759264</td>\n",
       "      <td>7.485170</td>\n",
       "      <td>41</td>\n",
       "      <td>Fibroblasts</td>\n",
       "      <td>CS171</td>\n",
       "      <td>H38</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>scRnaSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTCGATTCGATAACC-41</th>\n",
       "      <td>-4.757702</td>\n",
       "      <td>6.576497</td>\n",
       "      <td>41</td>\n",
       "      <td>Fibroblasts</td>\n",
       "      <td>CS171</td>\n",
       "      <td>H38</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>scRnaSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGGAGTCCACTGGG-41</th>\n",
       "      <td>-4.283028</td>\n",
       "      <td>7.481643</td>\n",
       "      <td>41</td>\n",
       "      <td>Fibroblasts</td>\n",
       "      <td>CS171</td>\n",
       "      <td>H38</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>scRnaSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>167598 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       UMAP_1    UMAP_2  cluster                    annot  \\\n",
       "cell                                                                        \n",
       "AAACCTGTCAGGATCT-1   0.738062 -3.097089       34  Mono+mono derived cells   \n",
       "AAACGGGAGCACAGGT-1   0.367818 -3.251462       34  Mono+mono derived cells   \n",
       "AAACGGGAGTGTGGCA-1  -0.728939 -5.994781       14  Mono+mono derived cells   \n",
       "AAACGGGAGTTAGGTA-1   2.814624 -7.131484        7  Mono+mono derived cells   \n",
       "AAACGGGCACCAGATT-1   3.420414 -6.966918        7  Mono+mono derived cells   \n",
       "...                       ...       ...      ...                      ...   \n",
       "TTTACCAAGAGAGGTA-41 -3.674058  7.171934       41              Fibroblasts   \n",
       "TTTACCAGTATCGAAA-41 -4.356690  7.339135       41              Fibroblasts   \n",
       "TTTCCTCGTGAGGCAT-41 -3.759264  7.485170       41              Fibroblasts   \n",
       "TTTCGATTCGATAACC-41 -4.757702  6.576497       41              Fibroblasts   \n",
       "TTTGGAGTCCACTGGG-41 -4.283028  7.481643       41              Fibroblasts   \n",
       "\n",
       "                     sample patient  digest typeSample  diet  \n",
       "cell                                                          \n",
       "AAACCTGTCAGGATCT-1   CISE06     H02  exVivo    citeSeq  Lean  \n",
       "AAACGGGAGCACAGGT-1   CISE06     H02  exVivo    citeSeq  Lean  \n",
       "AAACGGGAGTGTGGCA-1   CISE06     H02  exVivo    citeSeq  Lean  \n",
       "AAACGGGAGTTAGGTA-1   CISE06     H02  exVivo    citeSeq  Lean  \n",
       "AAACGGGCACCAGATT-1   CISE06     H02  exVivo    citeSeq  Lean  \n",
       "...                     ...     ...     ...        ...   ...  \n",
       "TTTACCAAGAGAGGTA-41   CS171     H38  exVivo   scRnaSeq  Lean  \n",
       "TTTACCAGTATCGAAA-41   CS171     H38  exVivo   scRnaSeq  Lean  \n",
       "TTTCCTCGTGAGGCAT-41   CS171     H38  exVivo   scRnaSeq  Lean  \n",
       "TTTCGATTCGATAACC-41   CS171     H38  exVivo   scRnaSeq  Lean  \n",
       "TTTGGAGTCCACTGGG-41   CS171     H38  exVivo   scRnaSeq  Lean  \n",
       "\n",
       "[167598 rows x 9 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "\n",
    "metadata_new = pd.read_csv(\"C:/Users/Vaishnavi Gaad/Desktop/MS/Sem 2/Capstone Project/Testing/annot_humanAll.csv\", index_col=\"cell\")\n",
    "metadata_new\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "abc.obs = metadata_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UMAP_1</th>\n",
       "      <th>UMAP_2</th>\n",
       "      <th>cluster</th>\n",
       "      <th>annot</th>\n",
       "      <th>sample</th>\n",
       "      <th>patient</th>\n",
       "      <th>digest</th>\n",
       "      <th>typeSample</th>\n",
       "      <th>diet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAACCTGTCAGGATCT-1</th>\n",
       "      <td>0.738062</td>\n",
       "      <td>-3.097089</td>\n",
       "      <td>34</td>\n",
       "      <td>Mono+mono derived cells</td>\n",
       "      <td>CISE06</td>\n",
       "      <td>H02</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>citeSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACGGGAGCACAGGT-1</th>\n",
       "      <td>0.367818</td>\n",
       "      <td>-3.251462</td>\n",
       "      <td>34</td>\n",
       "      <td>Mono+mono derived cells</td>\n",
       "      <td>CISE06</td>\n",
       "      <td>H02</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>citeSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACGGGAGTGTGGCA-1</th>\n",
       "      <td>-0.728939</td>\n",
       "      <td>-5.994781</td>\n",
       "      <td>14</td>\n",
       "      <td>Mono+mono derived cells</td>\n",
       "      <td>CISE06</td>\n",
       "      <td>H02</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>citeSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACGGGAGTTAGGTA-1</th>\n",
       "      <td>2.814624</td>\n",
       "      <td>-7.131484</td>\n",
       "      <td>7</td>\n",
       "      <td>Mono+mono derived cells</td>\n",
       "      <td>CISE06</td>\n",
       "      <td>H02</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>citeSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACGGGCACCAGATT-1</th>\n",
       "      <td>3.420414</td>\n",
       "      <td>-6.966918</td>\n",
       "      <td>7</td>\n",
       "      <td>Mono+mono derived cells</td>\n",
       "      <td>CISE06</td>\n",
       "      <td>H02</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>citeSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTACCAAGAGAGGTA-41</th>\n",
       "      <td>-3.674058</td>\n",
       "      <td>7.171934</td>\n",
       "      <td>41</td>\n",
       "      <td>Fibroblasts</td>\n",
       "      <td>CS171</td>\n",
       "      <td>H38</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>scRnaSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTACCAGTATCGAAA-41</th>\n",
       "      <td>-4.356690</td>\n",
       "      <td>7.339135</td>\n",
       "      <td>41</td>\n",
       "      <td>Fibroblasts</td>\n",
       "      <td>CS171</td>\n",
       "      <td>H38</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>scRnaSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTCCTCGTGAGGCAT-41</th>\n",
       "      <td>-3.759264</td>\n",
       "      <td>7.485170</td>\n",
       "      <td>41</td>\n",
       "      <td>Fibroblasts</td>\n",
       "      <td>CS171</td>\n",
       "      <td>H38</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>scRnaSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTCGATTCGATAACC-41</th>\n",
       "      <td>-4.757702</td>\n",
       "      <td>6.576497</td>\n",
       "      <td>41</td>\n",
       "      <td>Fibroblasts</td>\n",
       "      <td>CS171</td>\n",
       "      <td>H38</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>scRnaSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGGAGTCCACTGGG-41</th>\n",
       "      <td>-4.283028</td>\n",
       "      <td>7.481643</td>\n",
       "      <td>41</td>\n",
       "      <td>Fibroblasts</td>\n",
       "      <td>CS171</td>\n",
       "      <td>H38</td>\n",
       "      <td>exVivo</td>\n",
       "      <td>scRnaSeq</td>\n",
       "      <td>Lean</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>167598 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       UMAP_1    UMAP_2  cluster                    annot  \\\n",
       "cell                                                                        \n",
       "AAACCTGTCAGGATCT-1   0.738062 -3.097089       34  Mono+mono derived cells   \n",
       "AAACGGGAGCACAGGT-1   0.367818 -3.251462       34  Mono+mono derived cells   \n",
       "AAACGGGAGTGTGGCA-1  -0.728939 -5.994781       14  Mono+mono derived cells   \n",
       "AAACGGGAGTTAGGTA-1   2.814624 -7.131484        7  Mono+mono derived cells   \n",
       "AAACGGGCACCAGATT-1   3.420414 -6.966918        7  Mono+mono derived cells   \n",
       "...                       ...       ...      ...                      ...   \n",
       "TTTACCAAGAGAGGTA-41 -3.674058  7.171934       41              Fibroblasts   \n",
       "TTTACCAGTATCGAAA-41 -4.356690  7.339135       41              Fibroblasts   \n",
       "TTTCCTCGTGAGGCAT-41 -3.759264  7.485170       41              Fibroblasts   \n",
       "TTTCGATTCGATAACC-41 -4.757702  6.576497       41              Fibroblasts   \n",
       "TTTGGAGTCCACTGGG-41 -4.283028  7.481643       41              Fibroblasts   \n",
       "\n",
       "                     sample patient  digest typeSample  diet  \n",
       "cell                                                          \n",
       "AAACCTGTCAGGATCT-1   CISE06     H02  exVivo    citeSeq  Lean  \n",
       "AAACGGGAGCACAGGT-1   CISE06     H02  exVivo    citeSeq  Lean  \n",
       "AAACGGGAGTGTGGCA-1   CISE06     H02  exVivo    citeSeq  Lean  \n",
       "AAACGGGAGTTAGGTA-1   CISE06     H02  exVivo    citeSeq  Lean  \n",
       "AAACGGGCACCAGATT-1   CISE06     H02  exVivo    citeSeq  Lean  \n",
       "...                     ...     ...     ...        ...   ...  \n",
       "TTTACCAAGAGAGGTA-41   CS171     H38  exVivo   scRnaSeq  Lean  \n",
       "TTTACCAGTATCGAAA-41   CS171     H38  exVivo   scRnaSeq  Lean  \n",
       "TTTCCTCGTGAGGCAT-41   CS171     H38  exVivo   scRnaSeq  Lean  \n",
       "TTTCGATTCGATAACC-41   CS171     H38  exVivo   scRnaSeq  Lean  \n",
       "TTTGGAGTCCACTGGG-41   CS171     H38  exVivo   scRnaSeq  Lean  \n",
       "\n",
       "[167598 rows x 9 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abc.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "abc.write(\"Liver_Cell_Atlas.h5ad\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
